### PR TITLE
Selective invoice finance loan writeback

### PR DESCRIPTION
**Pull request by dcoplowe**
# Description

* Include a summary of the new content you're adding or the reason for the change. 
* Include relevant context.
* Do not link to work items.
* Any PNG images should be compressed using [Tiny PNG](https://tinypng.com/) or equivalent.

## Type of change

Please delete options that are not relevant.

- [x] New document(s)/updating existing
- [ ] Fixes
- [ ] Styling
- [ ] Bug fix (non-breaking change which fixes an issue)

## Reviews and merging

You are responsible for getting your PR merged. Address review comments promptly and **make sure to merge the PR when ready**.
Feel free to 'Enable automerge' - your PR will automatically merge when accepted and passing the build.

### CODE REVIEW

File: docs/lending/guides/loan-writeback/configure.md [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+</Tabs>
+
+Display the response to the customer and allow them to select the account. Store the returned bank account as `borrowersBankAccount` and use it to access properties on the borrower's bank account in future operations. 
+
+For the lender's bank account, use the `lendersBankAccount` value you would have created when implementing [bank feeds](../terms/bank-feed).
+
+### Supplier
+
+In order to create a *spend money* transaction, Codat requires you, the lender, to be represented as a [supplier](../terms/supplier) in your SMB's accounting system.
```
Comment (mcclowes) [inline docs/lending/guides/loan-writeback/configure.md:133]:
```suggestion
In order to create a *spend money* transaction, Codat requires you, the lender, to be represented as a [supplier](../../../terms/supplier) in your SMB's accounting system. 
```

try this

you're saying:
- go back out of /loan-writeback
- go back out of /guides
- go back out of /lending
- go into terms..

(I think)

File: docs/lending/guides/loan-writeback/deposit.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Once you receive the configuration information, you are ready to deposit funds into the borrower's bank account. This is known as *loan drawdown* general lending, and an *advance* in invoice finance. You will need to: 
+
+1. [Create a transfer](/lending/guides/loan-writeback/deposit#create-transfer) from the lender's bank account to the borrower's bank account. 
+
+2. [Create a bank transaction](/lending/guides/loan-writeback/deposit#create-bank-transaction) depositing the amount into the lender's bank account to make the funds available for drawdown.
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:14]:
```suggestion
2. [Create a bank feed transaction](/lending/guides/loan-writeback/deposit#create-bank-feed-transaction) to represent the account transfer in the lenders bank account. 
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+```mermaid
+    sequenceDiagram
+        participant backend as Your application 
+        participant codat as Codat
+
+        backend ->> codat: Create transfer (loan drawdown, lender -> bank account)
+        codat -->> backend:  transfer
+
+        backend ->> codat: Create bank transaction (deposit to lender's account)
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:25]:
```suggestion
        backend ->> codat: Create bank feed transaction (deposit to lender's account)
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+<TabItem value="python" label="Python">
+
+```python
+bank_transactions_create_request = operations.CreateBankTransactionsRequest(
+    accounting_create_bank_transactions=shared.AccountingCreateBankTransactions(
+        account_id=lenders_bank_account.id,
+        transactions=[
+            shared.CreateBankAccountTransaction(
+                id=transaction_id, # Unique identifier for this bank transaction
+                amount=Decimal(deposit_amount),
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:239]:
```suggestion
                amount=Decimal(-deposit_amount),
```

File: docs/lending/guides/loan-writeback/deposit.md (hunk 3 above)
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:25]:
```suggestion
        backend ->> codat: Create bank transaction (withdrawal from lender's account)
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+<TabItem value="nodejs" label="TypeScript">
+
+```javascript
+codatLending.loanWriteback.bankTransactions.create({
+    accountingCreateBankTransactions: {
+        accountId: lendersBankAccount.id,
+        transactions: [
+        {
+            id: transactionId, // Unique identifier for this bank transaction
+            amount: depositAmount,
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:213]:
```suggestion
            amount: -depositAmount,
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+```
+
+#### Request body
+
+```json
+{
+  "accountId": lendersBankAccount.id,
+  "transactions": [{
+    "id": transactionId, // Unique identifier for this bank transaction
+    "amount": depositAmount,
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:312]:
```suggestion
    "amount": -depositAmount,
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+    }
+}
+```
+
+</TabItem>
+
+</Tabs>
+
+
+### Create bank transaction
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:193]:
```suggestion
### Create bank feed transaction
```

File: docs/lending/guides/loan-writeback/repay-general-loan.md [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+        application ->> codat: Create transfer (bank account -> lender)
+        codat -->> application:  transfer
+    end
+    
+    alt paying interest and/or fees
+        application ->> codat: Create direct cost
+        codat -->> application: direct cost
+    end
+
+    application ->> codat: Create bank transactions (deposit in lender's account)
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-general-loan.md:39]:
```suggestion
    application ->> codat: Create bank feed transactions (deposit in lender's account)
```

File: docs/lending/guides/loan-writeback/repay-general-loan.md [hunk 9]
```diff
@@ -0,0 +1,567 @@
...
+			"id": expenseAccount.id
+		}
+	}]
+}
+```
+</TabItem>
+
+</Tabs>
+
+### Create bank transaction
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-general-loan.md:430]:
```suggestion
### Create bank feed transactions
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 10]
```diff
@@ -0,0 +1,827 @@
...
+
+To reflect that programmatically, perform these steps:
+
+1. [Create a transfer](/lending/guides/loan-writeback/repay-inoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.
+
+2. To record interest or fees, [create a direct cost](/lending/guides/loan-writeback/repay-inoice-finance#create-direct-cost) against the lender's bank account.
+
+3. To update the invoice as paid, [create a payment](/lending/guides/loan-writeback/repay-inoice-finance#create-payment) in the lender's bank account.
+
+4. [Create bank transactions](/lending/guides/loan-writeback/repay-inoice-finance#create-bank-transactions) to keep the lender's bank account transactions in sync.
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:21]:
```suggestion
4. [Create bank feed transactions](/lending/guides/loan-writeback/repay-invoice-finance#create-bank-feed-transactions) to represent the transfer, direct cost and invoice payment in the lender's bank account.
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 11]
```diff
@@ -0,0 +1,827 @@
...
+
+Once the borrower's customer has paid for the goods or services they purchased, the financed invoice is ready to be reconciled in the accounting platform.  
+
+To reflect that programmatically, perform these steps:
+
+1. [Create a transfer](/lending/guides/loan-writeback/repay-inoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.
+
+2. To record interest or fees, [create a direct cost](/lending/guides/loan-writeback/repay-inoice-finance#create-direct-cost) against the lender's bank account.
+
+3. To update the invoice as paid, [create a payment](/lending/guides/loan-writeback/repay-inoice-finance#create-payment) in the lender's bank account.
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:19]:
```suggestion
3. To update the invoice as paid, [create a payment](/lending/guides/loan-writeback/repay-invoice-finance#create-payment) in the lender's bank account.
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 12]
```diff
@@ -0,0 +1,827 @@
...
+import TabItem from '@theme/TabItem';
+import WritebackMapping from "@components/global/Prototypes/WritebackMapping";
+
+Once the borrower's customer has paid for the goods or services they purchased, the financed invoice is ready to be reconciled in the accounting platform.  
+
+To reflect that programmatically, perform these steps:
+
+1. [Create a transfer](/lending/guides/loan-writeback/repay-inoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.
+
+2. To record interest or fees, [create a direct cost](/lending/guides/loan-writeback/repay-inoice-finance#create-direct-cost) against the lender's bank account.
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:17]:
```suggestion
2. To record interest or fees, [create a direct cost](/lending/guides/loan-writeback/repay-invoice-finance#create-direct-cost) against the lender's bank account.
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 13]
```diff
@@ -0,0 +1,827 @@
...
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import WritebackMapping from "@components/global/Prototypes/WritebackMapping";
+
+Once the borrower's customer has paid for the goods or services they purchased, the financed invoice is ready to be reconciled in the accounting platform.  
+
+To reflect that programmatically, perform these steps:
+
+1. [Create a transfer](/lending/guides/loan-writeback/repay-inoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:15]:
```suggestion
1. [Create a transfer](/lending/guides/loan-writeback/repay-invoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 14]
```diff
@@ -0,0 +1,827 @@
...
+        backend ->> codat: Create transfer (lender -> bank account)
+        codat -->> backend:  transfer
+
+        backend ->> codat: Create direct cost
+        codat -->> backend: direct cost
+        
+        backend ->> codat: Create payment (on invoice)
+        codat -->> backend: payment
+        
+        backend ->> codat: Create bank transactions (transfer, direct cost and invoice payment amounts)
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:37]:
```suggestion
        backend ->> codat: Create bank feed transactions (transfer, direct cost and invoice payment amounts)
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 15]
```diff
@@ -0,0 +1,827 @@
...
+      "allocatedOnDate": invoicePaymentDate
+    }
+  ],
+}
+```
+</TabItem>
+
+</Tabs>
+
+### Create bank transactions
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:628]:
```suggestion
### Create bank feed transactions
```

File: docs/lending/guides/loan-writeback/repay-invoice-finance.md [hunk 16]
```diff
@@ -0,0 +1,827 @@
...
+        codat -->> backend:  transfer
+
+        backend ->> codat: Create direct cost
+        codat -->> backend: direct cost
+        
+        backend ->> codat: Create payment (on invoice)
+        codat -->> backend: payment
+        
+        backend ->> codat: Create bank transactions (transfer, direct cost and invoice payment amounts)
+        codat -->> backend: bank transactions
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-invoice-finance.md:38]:
```suggestion
        codat -->> backend: bank feed transactions
```

File: docs/lending/guides/loan-writeback/repay-general-loan.md [hunk 17]
```diff
@@ -0,0 +1,567 @@
...
+        codat -->> application:  transfer
+    end
+    
+    alt paying interest and/or fees
+        application ->> codat: Create direct cost
+        codat -->> application: direct cost
+    end
+
+    application ->> codat: Create bank transactions (deposit in lender's account)
+    codat -->> application: bank transactions
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/repay-general-loan.md:40]:
```suggestion
    codat -->> application: bank feed transactions
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 18]
```diff
@@ -0,0 +1,327 @@
...
+```mermaid
+    sequenceDiagram
+        participant backend as Your application 
+        participant codat as Codat
+
+        backend ->> codat: Create transfer (loan drawdown, lender -> bank account)
+        codat -->> backend:  transfer
+
+        backend ->> codat: Create bank transaction (deposit to lender's account)
+        codat -->> backend: bank transaction
```
Comment (dcoplowe) [inline docs/lending/guides/loan-writeback/deposit.md:26]:
```suggestion
        codat -->> backend: bank feed transaction
```

File: sidebars/lending.js [hunk 19]
```diff
@@ -76,13 +76,24 @@ module.exports = [
...
+  collapsed: true,
+  items: [
+    {
+      type: "doc",
+      id: "lending/guides/loan-writeback/introduction",
+      label: "Introduction",
+    },
+    "lending/guides/loan-writeback/configure",
+    "lending/guides/loan-writeback/deposit",
+    "lending/guides/loan-writeback/repay-general-loan",
```
Comment (dcoplowe) [inline sidebars/lending.js:94]:
```suggestion
    "lending/guides/loan-writeback/record-general-loan",
```

File: sidebars/lending.js [hunk 20]
```diff
@@ -76,13 +76,24 @@ module.exports = [
...
+    "lending/guides/loan-writeback/repay-invoice-finance",
```
Comment (dcoplowe) [inline sidebars/lending.js:95]:
```suggestion
    "lending/guides/loan-writeback/record-invoice-finance",
```

File: docs/lending/guides/loan-writeback/deposit.md [hunk 21]
```diff
@@ -0,0 +1,327 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Once you receive the configuration information, you are ready to deposit funds into the borrower's bank account. This is known as *loan drawdown* general lending, and an *advance* in invoice finance. You will need to: 
+
+1. [Create a transfer](/lending/guides/loan-writeback/deposit#create-transfer) from the lender's bank account to the borrower's bank account. 
+
+2. [Create a bank feed transaction](/lending/guides/loan-writeback/deposit#create-bank-feed-transaction) to represent the account transfer in the lenders bank account.
```
Comment (pzaichkina) [inline docs/lending/guides/loan-writeback/deposit.md:14]:
```suggestion
2. [Create a bank feed transaction](/lending/guides/loan-writeback/deposit#create-bank-feed-transaction) to represent the account transfer in the lender's bank account. 
```

**Review by pzaichkina — APPROVED** [review]

**Pull request merged**